json_test.go

```diff
@@ -42,12 +45,28 @@ func TestJSONParser(t *testing.T) {
 			file: "testdata/bad.json",
 			want: fftest.Vars{WantParseErrorIs: io.ErrUnexpectedEOF},
 		},
+		{
+			vars: fftest.NestedDefaultVars("."),
+			name: "nested objects",
+			file: "testdata/nested.json",
+			want: fftest.Vars{S: "a string", B: true, I: 123, F: 1.23, X: []string{"one", "two", "three"}},
+		}, {
+			vars: fftest.NestedDefaultVars("-"),
+			name: "nested objects hyphen delimiter",
+			file: "testdata/nested.json",
+			want: fftest.Vars{S: "a string", B: true, I: 123, F: 1.23, X: []string{"one", "two", "three"}},
+			opts: []ff.JSONOption{ff.WithJSONDelimiter("-")},
+		},
```

Comment:
Ugh, no, I see the issue here, and I don't think this change to the test actually works. Give me a moment, I'll suggest an alternative approach.

Reply:
Combined with the `vars` above this should set the flags and delimiter properly for parsing the given document, no?

Reply:
Sorry, I mis-spoke; the change here clearly "works" in the sense that the tests pass. My issue is that the flag names in NestedDefaultVars are tightly coupled to the hierarchy expressed in nested.json, which is somewhat arbitrary and fragile. I think there is  a more robust solution, which I'll suggest here in a bit